tsdb/engine/tsm1/wal.go

```diff
@@ -91,13 +91,12 @@ type WAL struct {
 
 func NewWAL(path string) *WAL {
 	db, rp := tsdb.DecodeStorePath(path)
-	return &WAL{
+	w := &WAL{
```

Comment:
Since you're just setting the `logger` to `os.Stderr` I don't see the advantage of making `NewWAL` two lines longer, you can just leave the previous `logger:      log.New(os.Stderr, "[tsm1wal] ", log.LstdFlags)` field setting.